src/main/java/org/opentripplanner/updater/stoptime/TimetableSnapshotSource.java

```diff
@@ -902,6 +909,20 @@ private boolean validateAndHandleModifiedTrip(TimetableSnapshotSourceMetrics met
         return success;
     }
 
+    /**
+     * Set tripheadsign from a realtime update
+     * @param tripUpdate
+     * @param trip
+     */
+    public static void setTripHeadsignFromUpdate(TripUpdate tripUpdate, Trip trip) {
+        if (tripUpdate.hasExtension(GtfsRealtimeOneBusAway.obaTripUpdate)) {
+            GtfsRealtimeOneBusAway.OneBusAwayTripUpdate updateExtension= tripUpdate.getExtension(GtfsRealtimeOneBusAway.obaTripUpdate);
+            if (!updateExtension.getTripHeadsign().isEmpty()) {
```

Comment:
does updateExtensions provide a hasTripHeadsign() method?  If so, use that, as calling isEmpty() may generate a NullPointerException.